web-app/src/games/api.games.controller.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  Body,
+  Controller,
+  Post,
+  Put,
+  Get,
+  Param,
+  Req,
+  UseGuards,
+  NotFoundException,
+} from '@nestjs/common';
+import { Request } from 'express';
+import { Coordinates } from '../common/utils';
+import { GamesService, GameAndStatus } from './games.service';
+import { Game } from './game.entity';
+import { AuthGuard } from '@nestjs/passport';
+
+@UseGuards(AuthGuard('jwt'))
+@Controller('api/games')
+export class ApiGamesController {
+  constructor(private readonly gamesService: GamesService) { }
+
+  @Post('')
+  async create(
+    @Body() params: { size?: number },
+    @Req() req: Request
+  ): Promise<Game> {
+    return await this.gamesService.createNewGame(params.size, req.user["username"]);
+  }
+
+  @Get(':id')
+  async findOne(
+    @Param('id') id: number,
+    @Req() req: Request
+  ): Promise<GameAndStatus> {
+    const game = await this.gamesService.findGameById(id);
+    if (game === undefined) throw new NotFoundException();
+    return this.gamesService.getGameAndStatus(game, req.user["username"]);
+  }
+
+  @Put(':id')
+  async update(
+    @Param('id') id: number,
+    @Body() params: { nextMove: Coordinates },
+    @Req() req: Request
+  ): Promise<GameAndStatus> {
+    let game = await this.gamesService.findGameById(id);
+    if (game === undefined) throw new NotFoundException();
+    if (params.nextMove.x && params.nextMove.y) {
+      game = await this.gamesService.updateGameState(game, {
+        x: params.nextMove.x,
+        y: params.nextMove.y,
+      });
+    }
+    return this.gamesService.getGameAndStatus(game, req.user["username"]);
+  }
+
+  @Get(':id/join')
+  async joinGame(
+    @Param('id') id: number,
+    @Req() req: Request) {
+    await this.gamesService.joinGame(id, req.user["username"]);

```

Comment:
It would be great not to have to copy `req.user["username"]` everytime, but I don't know how we could avoid it... If s.o. knows, I'd be curious 🙂

Reply:
The best I can do for now is to use the new type coming from https://github.com/marmelab/Hex/pull/47 !
```ts
@Get(':id/join')
  async joinGame(
    @Param('id') id: number,
    @Request() req: { user: UserDataInjectedInRequestAfterAuth }
  ) {
    const game = await this.gamesService.findGameById(id);
    if (game === undefined) throw new NotFoundException();
    await this.gamesService.joinGame(id, req.user.username);
  }
```